packages/firestore/exp/src/api/database.ts

```diff
@@ -247,7 +254,9 @@ export function clearIndexedDbPersistence(
   firestore: firestore.FirebaseFirestore
 ): Promise<void> {
   const firestoreImpl = cast(firestore, Firestore);
-  return firestoreImpl._clearPersistence();
+  return firestoreImpl._clearPersistence((databaseId, persistenceKey) => {
+    return clearPersistence(buildStoragePrefix(databaseId, persistenceKey));
```

Comment:
This code reads as if it's circular unless you pay careful attention. If `clearPersistence` were named `indexedDbClearPersistence` it would be clearer that the higher level API was delegating to the lower-level implementation.

I realize this is getting at that type-operation naming discussion we started earlier so feel free to defer if you must, but I think it's really hurting the ability to understand what's going on here while reading the code directly. Sure you can go look at the imports but that breaks flow. Reading code is hard enough and we should strive to make it easy to understand.

Reply:
I stumbled across this earlier too, but didn't act upon my confusion. I think a minimum bar for code quality should be that at the very least the person writing it shouldn't be confused...

Renamed to `indexedDbClearPersistence` and `indexedDbStoragePrefix`.